airflow/providers/postgres/hooks/postgres.py

```diff
@@ -320,6 +320,34 @@ def _generate_insert_sql(
 
         return sql
 
+    def ingest_embedding(self, table: str, input_data: Iterable[tuple[str, list[float]]], vector_size: int) -> None:
+        """
+        Store embedding vector in Postgres table.
+
+        :param table: The Name of the table
+        :param input_data: Iterable containing tuples of input data and corresponding embedding vectors.
+        :param vector_size: The size of vector. The maximum dimensions can be 2,000
+        """
+        from pgvector.psycopg import register_vector
+        from psycopg2 import sql
+
+        self.conn.execute("CREATE EXTENSION IF NOT EXISTS vector")
+        register_vector(self.conn)
+
+        create_table_query = sql.SQL(
+            "CREATE TABLE IF NOT EXISTS {} (id bigserial PRIMARY KEY, content text, embedding vector({}))"
+        ).format(sql.Identifier(table), sql.Literal(vector_size))
+
+        self.conn.execute(create_table_query)
+
+        for data_item in input_data:
+            insert_query = sql.SQL("INSERT INTO {} (content, embedding) VALUES (%s, %s)").format(
+                sql.Identifier(table)
+            )
+            content = data_item[0]
+            embedding = data_item[1]
+            self.conn.execute(insert_query, (content, embedding))
```

Comment:
```suggestion
            self.conn.execute(insert_query, data)
```

The unpacking seems unnecessary since the two are joined immediately?